### ISSUE TITLE
Worker thread support for "working" icon indicator

### ISSUE THREAD

**Issue by mjherzog** [aboutcode-org/scancode-workbench#552]
When accessing large Scan files it is often hard to determine if SCWB is "working" or hung in some error state. We need some "working" icon to indicate when SCWB is actually processing a command.

**Comment by OmkarPh** [comment]
whenever, we perform a sqlite query or heavy stuff, it should show this working text & some icon ig, in a corner or bottom of the screen right

**Comment by AyanSinhaMahapatra** [comment]
>  we perform a sqlite query or heavy stuff

Yes, this could be filters or searches, so basically queries. 

> working text & some icon ig, in a corner or bottom of the screen right

Just an icon is fine (usually gears moving is used for this?), we can determine where to put it from other similar tools/UI from popular tools, I'll look into this too.
Options are:

1. change the mouse pointer to the working icon
2. have this in a corner/edge
3. have an overlay (partially transparent) with this icon
or something else

**Comment by OmkarPh** [comment]
I think having it in a corner would be a good option
For smaller queries, the overlay would just flash in and out quickly (looks unpleasant)

**Comment by AyanSinhaMahapatra** [comment]
> I think having it in a corner would be a good option
> For smaller queries, the overlay would just flash in and out quickly (looks unpleasant)

Yeah sounds good, finding the right size would also be nice, so that it's visible but not too unpleasant

**Comment by OmkarPh** [comment]
The working indicator works fine for most of the scans

For very large scans and lesser computer resources, the indicator may feel unresponsive, 
It can be fixed by using worker threads as experimented in [feature/workingIndicator-webworkers-test](https://github.com/nexB/scancode-workbench/tree/feature/workingIndicator-webworkers-test)

Implementing & integrating web workers is a lot more work, and may be worked upon in the future.
For now, it's not very critical, Hence renaming the issue for further improvement
